src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerTaskConsumer.cs

```diff
@@ -101,49 +101,57 @@ public async Task<TaskInfo> CompleteAsync(string taskId, TaskResultData taskResu
 
-                    TaskInfo taskInfo = new TaskInfo()
+                    var taskInfoTable = VLatest.TaskInfo;
+                    while (sqlDataReader.Read())
                     {
-                        TaskId = id,
-                        QueueId = queueId,
-                        Status = (TaskStatus)status,
-                        TaskTypeId = taskTypeId,
-                        RunId = taskRunId,
-                        IsCanceled = isCanceled,
-                        RetryCount = retryCount,
-                        MaxRetryCount = maxRetryCount,
-                        HeartbeatDateTime = heartbeatDateTime,
-                        InputData = inputData,
-                        Context = taskContext,
-                        Result = result,
-                    };
+                        string id = sqlDataReader.Read(taskInfoTable.TaskId, 0);
+                        _ = sqlDataReader.Read(taskInfoTable.QueueId, 1);
+                        short status = sqlDataReader.Read(taskInfoTable.Status, 2);
+                        short taskTypeId = sqlDataReader.Read(taskInfoTable.TaskTypeId, 3);
+                        string taskRunId = sqlDataReader.Read(taskInfoTable.RunId, 4);
+                        bool isCanceled = sqlDataReader.Read(taskInfoTable.IsCanceled, 5);
+                        short retryCount = sqlDataReader.Read(taskInfoTable.RetryCount, 6);
+                        short maxRetryCount = sqlDataReader.Read(taskInfoTable.MaxRetryCount, 7);
+                        DateTime? heartbeatDateTime = sqlDataReader.Read(taskInfoTable.HeartbeatDateTime, 8);
+                        string inputData = sqlDataReader.Read(taskInfoTable.InputData, 9);
+                        string taskContext = sqlDataReader.Read(taskInfoTable.TaskContext, 10);
+                        string result = sqlDataReader.Read(taskInfoTable.Result, 11);
+
+                        TaskInfo taskInfo = new TaskInfo()
+                        {
+                            TaskId = id,
+                            QueueId = queueId,
+                            Status = (TaskStatus)status,
+                            TaskTypeId = taskTypeId,
+                            RunId = taskRunId,
+                            IsCanceled = isCanceled,
+                            RetryCount = retryCount,
+                            MaxRetryCount = maxRetryCount,
+                            HeartbeatDateTime = heartbeatDateTime,
+                            InputData = inputData,
+                            Context = taskContext,
+                            Result = result,
+                        };
 
-                    output.Add(taskInfo);
+                        output.Add(taskInfo);
+                    }
                 }
             }
+            catch (SqlException e) when (e.Message.Contains("Could not find stored procedure", StringComparison.OrdinalIgnoreCase))
```

Comment:
I am wondering if we have sqlErrorCode for this exception and can we use that instead?

Reply:
It makes sense to check against sqlErrorCode, currently checking with a hardcoded value. Will get the changes in with a separate PR.